src/data/monoid_algebra.lean

```diff
@@ -119,40 +119,40 @@ lemma one_def : (1 : monoid_algebra k G) = single 1 1 :=
 rfl
 
 -- TODO: the simplifier unfolds 0 in the instance proof!
-private lemma zero_mul (f : monoid_algebra k G) : 0 * f = 0 :=
+protected lemma zero_mul (f : monoid_algebra k G) : 0 * f = 0 :=
 by simp only [mul_def, sum_zero_index]
 
-private lemma mul_zero (f : monoid_algebra k G) : f * 0 = 0 :=
+protected lemma mul_zero (f : monoid_algebra k G) : f * 0 = 0 :=
```

Comment:
I'm not sure it's worth sacrificing `private` for protected here and elsewhere. I don't think there's an easy way to do both. You could just call it `_aux` instead.

Reply:
What's the downside of `protected` over `private`?

Reply:
`private` lemmas aren't accessible outside of this file. It's not that big a deal, but I guess it just gives you a shorter list to search through after Ctrl-space.

Reply:
I see. Maybe in lean 3.16, we can combine `protected` and `private`? @gebner

Reply:
`private` and `protected` are completely different things?  I don't see why you would want to unify them?

Reply:
@gebner Not unify them, but currently you can only use one, even though they are orthogonal.
As far as I know, I can't write
```lean
private protected lemma mul_zero ...
```

Reply:
they are mutually exclusive in the sense that they both do different funny things to the namespace. `private` puts it in a `_private.123` namespace, so if you protected it as well you would never be able to refer to it.

Reply:
Hmm, that's unfortunate.

Reply:
These lemmas are here because Lean unfolds `(*)`. Could you please test if you can move their proofs to the next `instance` in Lean 3.15?

Reply:
@urkud I tried this. It didn't work.